flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -91,9 +91,12 @@
 
 	private static final Priority RM_REQUEST_PRIORITY = Priority.newInstance(1);
 
-	/** YARN container map. Package private for unit test purposes. */
+	/** YARN container map. */
 	private final ConcurrentMap<ResourceID, YarnWorkerNode> workerNodeMap;
 
+	/** Mapping the Container id to ResourceID. */
+	private final ConcurrentMap<ContainerId, ResourceID> containerIdToResourceIdMap;
```

Comment:
Why do we need to maintain this map? It seems to me that `ResourceId` can be deterministically generated from `ContainerId`.